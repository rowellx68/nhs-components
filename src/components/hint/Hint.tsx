import React from 'react';
import clsx from 'clsx';
import { ElementProps } from '@/types/shared';

export type HintProps = ElementProps<'div'>;

const Hint = ({ className, ...rest }: HintProps) => {
  return <div className={clsx('nhsuk-hint', className)} {...rest} />;
};

Hint.displayName = 'Hint';

export { Hint };
