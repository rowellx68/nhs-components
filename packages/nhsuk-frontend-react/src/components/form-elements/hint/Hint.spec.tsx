import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { Hint } from './Hint';

it('should render the hint component', () => {
  const { container } = render(
    <Hint>Use hint text to help users understand a question.</Hint>,
  );

  expect(container).toMatchSnapshot();
});
