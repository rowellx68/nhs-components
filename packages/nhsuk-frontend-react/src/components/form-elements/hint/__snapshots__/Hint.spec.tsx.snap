// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the hint component 1`] = `
<div>
  <div
    class="nhsuk-hint"
  >
    Use hint text to help users understand a question.
  </div>
</div>
`;
